compiler/src/main/kotlin/software/amazon/lastmile/kotlin/inject/anvil/processor/ContributesBindingProcessor.kt

```diff
@@ -118,17 +126,18 @@ internal class ContributesBindingProcessor(
                                     }
                                 }
                                 .apply {
-                                    val parameterName = clazz.innerClassNames().decapitalize()
-                                    addParameter(
-                                        ParameterSpec
-                                            .builder(
-                                                name = parameterName,
-                                                type = clazz.toClassName(),
-                                            )
-                                            .build(),
-                                    )
-
-                                    addStatement("return $parameterName")
+                                    val hasAssistedInjection = clazz.getConstructors()
+                                        .any { constructor ->
+                                            constructor.parameters.any {
+                                                it.isAnnotationPresent(Assisted::class)
```

Comment:
Note that this is not correct. While it makes no sense, you can technically have a constructor with an `@Assisted` parameter, but another constructor is annotated with `@Inject`. I'll add a test case for this.